core/src/stage.cpp

```diff
@@ -245,8 +245,10 @@ void StagePrivate::computeCost(const InterfaceState& from, const InterfaceState&
 	solution.setStartState(tmp_from);
 	solution.setEndState(tmp_to);
 
+	if (!cost_term_)
+		return;
+
```

Comment:
This patch does not make sense (and your "hotfix" comment agrees). If this refers to a container aggregation will not take place at all and the solution cost will stay 0.0. Which test failed for you?

The assert here is a bit fragile because [I default-assign the term only in `Stage::init()`](https://github.com/v4hn/moveit_task_constructor/commit/053a18a35c9b1a42a0589da0893a48eff23c0eeb#diff-f7194ce218ff51e4e6827b810d9c1b96R305). If the base class' init is not called the term will be unassigned here. I fixed a related issue in the tests already [here](https://github.com/v4hn/moveit_task_constructor/commit/b82f50587c39bb2c6768edcd89df84cfeb0ef0f6).